lib/src/java/facter.cc

```diff
@@ -0,0 +1,102 @@
+#include "com_puppetlabs_Facter.h"
+#include <facter/facts/collection.hpp>
+#include <facter/facts/scalar_value.hpp>
+#include <facter/facts/array_value.hpp>
+#include <facter/facts/map_value.hpp>
+#include <string>
+
+using namespace std;
+using namespace facter::facts;
+
+static string to_string(JNIEnv* env, jstring str)
+{
+    auto ptr = env->GetStringUTFChars(str, nullptr);
+    if (!ptr) {
+        return {};
```

Comment:
It's C++11 brace-initialization syntax.  It's equivalent to `return string()` here.